src/concrete/ml/sklearn/base.py

```diff
@@ -1281,19 +1287,67 @@ def __init_subclass__(cls):
                 _TREE_MODELS.add(cls)
                 _ALL_SKLEARN_MODELS.add(cls)
 
-    def __init__(self, n_bits: int):
+    def __init__(self, n_bits: Union[int, Dict[str, int]]):
         """Initialize the TreeBasedEstimatorMixin.
 
         Args:
-            n_bits (int): The number of bits used for quantization.
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and leaves. If a dict is
+                passed, then it should contain "op_inputs" and "op_leaves" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs (mandatory): number of bits to quantize the input values
+                    - op_leaves (optional): number of bits to quantize the leaves
+                Default to 6.
         """
-        self.n_bits: int = n_bits
+
+        # Check if 'n_bits' is a valid value.
+        _inspect_tree_n_bits(n_bits)
+
+        #: The number of bits to quantize the model.
+        self.n_bits: Union[int, Dict[str, int]] = n_bits
 
         #: The model's inference function. Is None if the model is not fitted.
         self._tree_inference: Optional[Callable] = None
 
+        #: Wether to perform the sum of the output's tree ensembles in FHE or not.
+        # By default, the decision of the tree ensembles is made in clear.
+        self._fhe_ensembling = False
+
         BaseEstimator.__init__(self)
 
+    @property
+    def fhe_ensembling(self) -> bool:
+        """Property getter for `_fhe_ensembling`.
```

Comment:
I don't understand why there is a "setter". This value should be set in the constructor, like `n_bits`. It should also be serialized

Reply:
As discussed in https://zama-ai.slack.com/archives/C0455QDGT6C/p1706087454383859

If the user alters a value of a private member outside of a class,  undefined behaviors are expected.